crates/teloxide/src/error_handlers.rs

```diff
@@ -23,6 +25,31 @@ where
     }
 }
 
+/// An asynchronous handler of an error with dependencies
+pub trait ErrorHandlerExt<E> {
```

Comment:
this makes no sense to me. 

1. why do we need an extension? `ErrorHandler` is in this crate, we can just change it
2. it isn't even an extension of `ErrorHandler` (extension usually means that a trait has a blanket impl for all types implementing the "base" trait)

Reply:
Initially I wanted to extend the `ErrorHandler` initially with the new method, but there is no way to implement such trait both for `Fn(E)` and `Fn(DependencyMap, E)`.

After that I tried to make the `ErrorHandlerExt` a super trait, but again to implement this for the function with the new signature it required the `Fn(DependencyMap, E)` to implement the `ErrorHandler`, so it's impossible currently, again.

If I just changed the signature of the `ErrorHandler`'s function, it will break to much existing code (especially in docs)

Reply:
Moreover, there is a place in the `Dispatcher`, when the `error_handler` is invoked for the `Update Listener` errors, so there is no much sense to use the `error_handler` with deps, so I would personally prefer to keep both functions, just not to pass an empty `DependencyMap` there

Reply:
I think we should go with the simple approach and just change the existing trait. Having two traits like this PR currently does is much more confusing.